File: Nov2020/21Nov/DailyTemperatures.java
/*
Problem Name: 739. Daily Temperatures
Problem Link: https://leetcode.com/problems/daily-temperatures/
*/
class Solution {
    public int[] dailyTemperatures(int[] T) {
        int ans[] = new int[T.length];
        int n = T.length;
        Stack<Integer>  st = new Stack<Integer>();
        for(int i=0;i<n;i++){
            while((!st.isEmpty()) && (T[st.peek()] < T[i])){
                int ind = st.pop();
                ans[ind] = i - ind;
            }
            st.push(i);
        }
        return ans;
    }
}

File: Nov2020/21Nov/DailyTemperatures.cpp
/*
Problem name: Daily Temperatures
Problem link: https://leetcode.com/problems/daily-temperatures/
*/

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        vector<int> days(T.size(),0);
        stack<int> s;
        for(int i=0;i<T.size();++i){
            while(!s.empty() && T[s.top()]<T[i]){
                days[s.top()]=i-s.top();
                s.pop();
            }
            s.push(i);
        }
        return days;
    }
};
